frontend/main.js

```diff
@@ -316,6 +332,33 @@ async function startApp () {
           })
         }
         window.favicon.badge(this.shouldSetBadge ? 1 : 0)
+      },
+      keepUpdated () {
```

Comment:
This file is getting very big, I would add a comment above this:

```js
// TODO: move this into a mixin to keep main.js small and separate 
//       out the functionality into individual update functions
```

Also, I think renaming this function to `periodicStateChecks` would make its purpose clearer.

Reply:
Okay.